### PR TITLE
Fix an e2e flake on updating node status

### LINKED ISSUE #91911: [Failing Test] Conformance - GCE - master (ci-kubernetes-gce-conformance-latest)
**Which jobs are failing**:

`Conformance - GCE - master (ci-kubernetes-gce-conformance-latest)`

**Which test(s) are failing**:

`[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Conformance]`

**Since when has it been failing**:

`06-08-20 07:22 PDT`

**Testgrid link**:

https://testgrid.k8s.io/sig-release-master-blocking#Conformance%20-%20GCE%20-%20master

**Reason for failure**:

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:597
Jun  8 15:43:46.663: Unexpected error:
    <*errors.StatusError | 0xc001b8c000>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {
                SelfLink: "",
                ResourceVersion: "",
                Continue: "",
                RemainingItemCount: nil,
            },
            Status: "Failure",
            Message: "Operation cannot be fulfilled on nodes \"bootstrap-e2e-minion-group-9n19\": the object has been modified; please apply your changes to the latest version and try again",
            Reason: "Conflict",
            Details: {
                Name: "bootstrap-e2e-minion-group-9n19",
                Group: "",
                Kind: "nodes",
                UID: "",
                Causes: nil,
                RetryAfterSeconds: 0,
            },
            Code: 409,
        },
    }
    Operation cannot be fulfilled on nodes "bootstrap-e2e-minion-group-9n19": the object has been modified; please apply your changes to the latest version and try again
occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduling/preemption.go:124
```

**Anything else we need to know**:

/sig scheduling
/cc @kubernetes/ci-signal
/priority critical-urgent
/milestone v1.19

Looks like changes were introduced here: https://github.com/kubernetes/kubernetes/pull/91243
/cc @cynepco3hahue

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind flake
/sig scheduling
/priority critical-urgent

**What this PR does / why we need it**:

Some scheduler preemption e2e assumes the call to update a node's status would always succeed. But it may conflict with other node status updating actors.

This PR replaces `UpdateStatus` with `patch` so as to reduce the possibility of conflicts - all preemption test would only update the node.status.capacity on a dedicated key.

**Which issue(s) this PR fixes**:

Fixes #91911

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
/assign @ahg-g 
cc/ @hasheddan

File: test/e2e/scheduling/preemption.go [hunk 1]
```diff
@@ -737,3 +736,21 @@ func waitForPreemptingWithTimeout(f *framework.Framework, pod *v1.Pod, timeout t
 	})
 	framework.ExpectNoError(err, "pod %v/%v failed to preempt other pods", pod.Namespace, pod.Name)
 }
+
+func patchNode(client clientset.Interface, old *v1.Node, new *v1.Node) error {
```
Comment (hasheddan) [inline test/e2e/scheduling/preemption.go:740]: I think this is an improvement, but I think the error described in the issue may be misleading as I am also seeing this: `Jun  9 12:22:18.370: failed pod observation expectations: rs1 had more than 2 pods created: 3`

Comment (Huang-Wei) [reply test/e2e/scheduling/preemption.go:740]: Are you referring to another test failure (other than `basic preemption works`)? That may be related (but I need to confirm) - the node is not updated successfully with proper "fake resource", hence the rs doesn't get expected replicas scheduled. But anyway, let focus on the `basic preemption works` failure in this PR.

Comment (hasheddan) [reply test/e2e/scheduling/preemption.go:740]: Ah yes, this new failure appears to be different. I will open a separate issue :+1:

**Review by hasheddan — APPROVED** [review]
/lgtm

/hold for @ahg-g

File: test/e2e/scheduling/preemption.go (hunk 1 above)
Comment (Huang-Wei) [reply test/e2e/scheduling/preemption.go:740]: Please, thanks - I'm aware of that one: "PreemptionExecutionPath runs ReplicaSets ". Hope this one fixes that one as well.

**Comment by Huang-Wei** [comment]
The verify script was complaining about:

https://github.com/kubernetes/kubernetes/blob/6ac3ca4b1793d0eaaaa7d605e5b9b8057000e1ce/test/e2e/scheduling/preemption.go#L243-L246

But this is pre-existing, so I'm just changing `Skipf` to `Fatalf` to work around CI. @oomichi @spiffxp please let me know if it's not legit.

File: test/e2e/scheduling/preemption.go [hunk 2]
```diff
@@ -204,8 +209,9 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 		pods := make([]*v1.Pod, 0, len(nodeList.Items))
 		for i, node := range nodeList.Items {
 			// Update each node to advertise 3 available extended resources
-			node.Status.Capacity[testExtendedResource] = resource.MustParse("3")
-			node, err := cs.CoreV1().Nodes().UpdateStatus(context.TODO(), &node, metav1.UpdateOptions{})
+			nodeCopy := node.DeepCopy()
+			nodeCopy.Status.Capacity[testExtendedResource] = resource.MustParse("3")
```
Comment (BenTheElder) [inline test/e2e/scheduling/preemption.go:213]: this seems like a rather [Disruptive] operation?

File: test/e2e/scheduling/preemption.go [hunk 3]
```diff
@@ -241,7 +247,7 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 			framework.Logf("Created pod: %v", pods[i].Name)
 		}
 		if len(pods) < 2 {
-			e2eskipper.Skipf("We need at least two pods to be created but" +
+			framework.Failf("We need at least two pods to be created but" +
```
Comment (spiffxp) [inline test/e2e/scheduling/preemption.go:250]: I'm fine with this change, it should have been this way to begin with.  This probably landed before the verify job started enforcing the no-skips requirement for conformance

**Review by spiffxp — APPROVED** [review]
/lgtm

File: test/e2e/scheduling/preemption.go (hunk 3 above)
Comment (oomichi) [reply test/e2e/scheduling/preemption.go:250]: Yeah, that is a nice point.
All conformance tests should be operated without considering any conditions.
So it seems fine to replace Skipf with Failf to know the conformance test is operated always.

BTW I found a small thing here.
We need to put a space after `but to distinguish it from later `all` word.

Comment (oomichi) [reply test/e2e/scheduling/preemption.go:250]: /lgtm

anyways :-)

**Pull request merged**